pytest_ansible/host_manager/__init__.py

```diff
@@ -0,0 +1,111 @@
+"""Fixme."""
+
+import ansible
+from pkg_resources import parse_version
+from pytest_ansible.logger import get_logger
+
+# conditionally import ansible libraries
+has_ansible_v2 = parse_version(ansible.__version__) >= parse_version('2.0.0')
+has_ansible_v24 = parse_version(ansible.__version__) >= parse_version('2.4.0')
+
+log = get_logger(__name__)
+
+
+class BaseHostManager(object):
+
+    """Pass."""
+
+    _required_kwargs = ('inventory',)
+
+    def __init__(self, *args, **kwargs):
+        self.options = kwargs
+
+        self.has_required_kwargs(**kwargs)
+
+        # Sub-classes should override this value
+        self._dispatcher = None
+
+        # Initialize ansible inventory manager
+        self.initialize_inventory()
+
+    def has_required_kwargs(self, **kwargs):
+        """Return whether the required kwargs were provided during instantiation."""
+        for kwarg in self._required_kwargs:
+            assert kwarg in self.options, "Missing required keyword argument '%s'" % kwarg
```

Comment:
Okay, `TypeError` will be raised.  And I've tweaked the function name and docstring to reflect reality.